src/settings/settingsdialog.cpp

```diff
@@ -310,6 +327,30 @@ void SettingsDialog::onOcrLanguagesPathChanged(const QString &path)
     ui->ocrLanguagesListWidget->addLanguages(Ocr::availableLanguages(path));
 }
 
+void SettingsDialog::onTesseractParametersCurrentItemChanged()
+{
+    if (ui->tesseractParametersTableWidget->currentRow() == -1) {
+        ui->tesseractParametersRemoveButton->setEnabled(false);
+    } else {
+        ui->tesseractParametersRemoveButton->setEnabled(true);
+    }
+}
+
+void SettingsDialog::onTesseractParametersAdd()
+{
+    ui->tesseractParametersTableWidget->addParameter();
+    int lastRow = ui->tesseractParametersTableWidget->rowCount() - 1;
+    QTableWidgetItem *newCell = ui->tesseractParametersTableWidget->item(lastRow, 0);
+    ui->tesseractParametersTableWidget->setCurrentItem(newCell);
+    ui->tesseractParametersTableWidget->editItem(newCell);
```

Comment:
All this code related to the `tesseractParametersTableWidget`. I would move this code inside `addParameter` and connect directly to it.